### ISSUE TITLE
Error for `Lint/MixedCaseRange` when invalid utf8 and offense exists

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12750]
## Expected behavior

Offense gets autocorrected.

## Actual behavior

RuboCop throws an error. This is because the range for the regexp literals does not contain the correct range.

## Steps to reproduce the problem

Note that `–` below is the "em dash", not the usual `-`.

```rb
it 'registers an offense for mixed range with invalid byte sequence in UTF-8' do
  expect_offense(<<~'RUBY'.sub(/\#{message}/, message))
    foo = /[\–A-z]/
              ^^^ #{message}
  RUBY

  expect_correction(<<~'RUBY')
    foo = /[\–A-Za-z]/
  RUBY
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.62.0 (using Parser 3.3.0.4, rubocop-ast 1.31.1, running on ruby 3.3.0) [x86_64-linux]
  - rubocop-performance 1.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.27.1
```

**Comment by Earlopain** [comment]
Can't reproduce anymore
